yarn-project/end-to-end/src/e2e_p2p/gossip_network_no_cheat.test.ts

```diff
@@ -0,0 +1,215 @@
+import type { Archiver } from '@aztec/archiver';
+import type { AztecNodeService } from '@aztec/aztec-node';
+import { EthAddress, sleep } from '@aztec/aztec.js';
+import { addL1Validator } from '@aztec/cli/l1';
+import { EthCheatCodesWithState } from '@aztec/ethereum/test';
+import { RollupAbi } from '@aztec/l1-artifacts/RollupAbi';
+import { StakingAssetHandlerAbi } from '@aztec/l1-artifacts/StakingAssetHandlerAbi';
+import type { SequencerClient } from '@aztec/sequencer-client';
+import { BlockAttestation, ConsensusPayload } from '@aztec/stdlib/p2p';
+
+import { jest } from '@jest/globals';
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import { getContract } from 'viem';
+
+import { shouldCollectMetrics } from '../fixtures/fixtures.js';
+import { type NodeContext, createNodes } from '../fixtures/setup_p2p_test.js';
+import { AlertChecker, type AlertConfig } from '../quality_of_service/alert_checker.js';
+import { P2PNetworkTest, SHORTENED_BLOCK_TIME_CONFIG, WAIT_FOR_TX_TIMEOUT } from './p2p_network.js';
+import { createPXEServiceAndSubmitTransactions } from './shared.js';
+
+const CHECK_ALERTS = process.env.CHECK_ALERTS === 'true';
+
+// Don't set this to a higher value than 9 because each node will use a different L1 publisher account and anvil seeds
+const NUM_NODES = 4;
+const NUM_TXS_PER_NODE = 2;
+const BOOT_NODE_UDP_PORT = 40600;
```

Comment:
try pick a port range different to the gossip test here. We get alot of addr already in use issues